sdv/metadata/single_table.py

```diff
@@ -280,28 +283,29 @@ def _determine_sdtype_for_objects(self, data):
             data (pandas.Series):
                 The data to be analyzed.
         """
-        sdtype = None
+        if len(data) <= 5:
+            sdtype = 'categorical'
+        else:
+            unique_values = data.nunique()
+            if unique_values == len(data):
+                sdtype = 'id'
+            elif unique_values <= round(len(data) / 5):
+                sdtype = 'categorical'
+            else:
+                sdtype = 'unknown'
+
         try:
             with warnings.catch_warnings():
                 warnings.simplefilter('ignore', category=UserWarning)
 
                 data_test = data.sample(10000) if len(data) > 10000 else data
                 datetime_format = get_datetime_format(data_test)
-                pd.to_datetime(data_test, format=datetime_format, errors='raise')
-
-            sdtype = 'datetime'
+                if datetime_format:
+                    pd.to_datetime(data_test, format=datetime_format, errors='raise')
```

Comment:
Is this the only line that raises? If so you could put the try/catch only around it. If that's not the case ignore this comment.

Reply:
I just want to be safe with `get_datetime_format` so I will shorten it there 👍🏻